keps/sig-auth/4317-pod-certificates/README.md

```diff
@@ -0,0 +1,1474 @@
+	  library function crypto/ecdsa.SignASN1)
+  * If the key is an ED25519 key, then the signature is as described by the
+    [ED25519 Specification](https://ed25519.cr.yp.to/) (as implemented by
+    the golang library crypto/ed25519.Sign).
+* Details of the requesting Pod's identity:
+  * Namespace (indicated by the namespace the PodCertificateRequest is created in).
+  * Pod name
+  * Pod UID
+  * Service account name
+  * Service account UID
+  * Node name
+  * Node UID
+* A requested maximum duration for the certificate lifetime.
+* The signer name from which a certificate is being requested.
+* (In the status subobject): An issued certificate chain, stored as a single
+  PEM-formatted string.  If there is more than one issued certificate, the first
+  element of the chain is assumed to be the leaf certificate, followed by any
+  necessary intermediates in leaf-to-root order.
+* (In the status subobject): A timestamp at which time the requester that made
+  the PodCertificateRequest should begin trying to refresh the certificate.
+
+PodCertificateRequest objects have a simplified lifecycle compared to
+CertificateSigningRequests.  There are no separate approval and signing stages.
+They can only be issued (and thus implicitly approved), denied (represented by a
+"Denied" status), or failed (represented by a "Failed" status).
+
+Similar to CertificateSigningRequests, controllers that want to issue a
+certificate in response to a PodCertificateRequest need to have two separate
+RBAC permissions:
+* `update` on `podcertificaterequests/status`.
+* `sign` on the requested signer.
+Similar to CertificateSigningRequest, we will also support wildcards on the RBAC signer permissions.
+
+Once a signer controller decides that a PodCertificateRequest is addressed to it
+by matching on the signer name, it should take one of the following actions:
+* Issue the certificate by updating the status subresource of the
+  PodCertificateRequest, setting both the `certificateChain` and
+  `beginRefreshAt` fields.
+* Reject the certificate by updating the status subresource of the
+  PodCertificateRequest, adding a "Denied" condition.
+* Indicate failure in issuing the certificate by updating the status subresource
+  of the PodCertificateRequest, adding a "Failed" condition.
+
+PodCertificateRequest validation logic will:
+* Confirm that the public key is one of the supported key types.
+* Confirm that the proof-of-possession is valid.
+* Confirm that the issued chain (if one is set) consists of valid certificates,
+  that do indeed form a chain to each other.
+* Confirm that neither or both of the certificateChain and beginRefreshAt
+  timestamp are set.
+* Ensure that all fields in PodCertificateRequest.Spec are immutable.
+* Ensure that all fields in PodCertificateRequest.Status are immutable once the
+  request has been issued, denied, or failed.
+
+The `maxExpirationSeconds` field will have the following logic:
```

Comment:
Probably need a min for this field -> does not make sense to let clients ask for 1 second certs.

Reply:
Good point.  Updated with a min of 1 hour.  It should be safe to drop that down in the future.